json2parquet/client.py

```diff
@@ -62,18 +63,30 @@ def _convert_data_with_column_names_dict(data, schema):
         schema_names.append(schema[column])
     return pa.RecordBatch.from_arrays(array_data, schema_names)
 
+def f7(seq):
+    seen = set()
+    seen_add = seen.add
+    return [x for x in seq if not (x in seen or seen_add(x))]
 
-def _convert_data_with_column_names(data, schema):
+
+def _convert_data_without_column_names(data):
     column_data = {}
     array_data = []
+    schema = []
+    row_count = 0
     for row in data:
+        names = list(row.keys())
+        schema = f7(schema + names)
+        row_count += 1
         for column in schema:
-            _col = column_data.get(column, [])
+            _col = column_data.get(column, [None]*(row_count-1))
```

Comment:
I'm adding as many None values in a list as the number of lines processed from the JSON file so far, so new columns will have as many values as the rest of the columns.